sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_models.py

```diff
@@ -191,7 +191,7 @@ class KeyVaultSecret(object):
     """All of a secret's properties, and its value."""
 
     def __init__(self, properties, value):
-        # type: (SecretProperties, str) -> None
+        # type: (SecretProperties, Optional[str]) -> None
```

Comment:
This is a case where the type checking gets a little confusing. A SecretBundle can have a `None` `value`, which is why this `Optional[str]` type makes sense -- but also, [SecretSetParameters](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_generated/v7_2/models/_models_py3.py#L560) requires a non-`None` value. We have to send over SecretSetParameters to create a secret in the first place, meaning that -- in theory -- any secret we get back and turn into a KeyVaultSecret _should_ have a value for `value`. I don't know what to do about this yet, but I'm thinking aloud a bit about why this is awkward.

Reply:
When does SecretBundle have a None value?

Reply:
It doesn't in any real scenario I've seen (and I doubt it ever would), but its [type signature](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_generated/v7_2/models/_models_py3.py#L128) leaves the possibility open.

Reply:
Then this introduces one question that happened several times in mypy ci: 
when swagger defines a property to be optional while we think there's no chance to be None, should we ignore mypy error in this case?